java_basic_programming/2019-10-18_SetAssignments/src/de/telran/Main.java

```diff
@@ -0,0 +1,44 @@
+package de.telran;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Main {
+
+    public static void main(String[] args) {
+	    List<String> names = new ArrayList<>();
+	    names.add("Ivan");
+        names.add("Maria");
+        names.add("Piotr");
+        names.add("Anna");
+        names.add("Maria");
+        names.add("Ivan");
+
+        List<String> dedupedNames = removeDuplicates(names);
+        System.out.println(dedupedNames);
+
+        List<String> duplicateNames = getDuplicates(names);
+        System.out.println(duplicateNames);
+    }
+
+    public static List<String> removeDuplicates(List<String> input) {
+        Set<String> set = new HashSet<>();
+        for (String string : input) {
+            set.add(string);
```

Comment:
There is an easier and shorter way to turn list into a set by using a constructor with a Collection as a parameter. No need to do this in the loop.